tarofreighter/chain_porter.go

```diff
@@ -1154,6 +1162,21 @@ func (p *ChainPorter) publishSubscriberEvent(event Event) {
 // chanutils.EventPublisher interface.
 var _ chanutils.EventPublisher[Event, bool] = (*ChainPorter)(nil)
 
+// ExecuteSendStateEvent is an event which is sent to the ChainPorter's event
+// subscribers before a state is executed.
+type ExecuteSendStateEvent struct {
+	// timeStamp is the time the event was created.
+	timeStamp time.Time
+
+	// sendState is the state that is about to be executed.
+	SendState SendState
```

Comment:
Perhaps the main interface should also allow access to this field?

Reply:
The doc is wrong here. Should be `SendState` rather than `sendState`. The field should be accessible on the struct as is. Should I wrap it in a method? I'm not sure what's best practice here.